velox/exec/tests/SpillTest.cpp

```diff
@@ -42,46 +43,156 @@ class SpillTest : public testing::Test,
+      }
+      state.setPartitionSpilled(partition);
+      EXPECT_TRUE(state.isPartitionSpilled(partition));
+      EXPECT_FALSE(state.hasFiles(partition));
+      for (auto batch = 0; batch < 10; ++batch) {
+        // We add a sorted run in two pieces: 1, 11, 21,,, followed by 100001 ,
+        // 100011, 100021   etc. where the last digit is the batch number. Each
+        // sorted run has 20000 rows.
+        state.appendToPartition(
+            partition,
+            makeRowVector({makeFlatVector<int64_t>(
+                10'000, [&](auto row) { return row * 10 + batch; })}));
+        EXPECT_TRUE(state.hasFiles(partition));
+
+        state.appendToPartition(
+            partition,
+            makeRowVector({makeFlatVector<int64_t>(10'000, [&](auto row) {
+              return row * 10 + batch + 100'000;
+            })}));
+        EXPECT_TRUE(state.hasFiles(partition));
+
+        // Indicates that the next additions to 'partition' are not sorted with
+        // respect to the values added so far.
+        state.finishWrite(partition);
+        EXPECT_TRUE(state.hasFiles(partition));
+        ++expectedSpilledFiles;
+      }
+    }
+    EXPECT_LT(200'000 * sizeof(int64_t), state.spilledBytes());
+    EXPECT_EQ(expectedSpilledFiles, state.spilledFiles());
+    spilledFiles = state.TEST_spilledFiles();
+    std::unordered_set<std::string> spilledFileSet(
+        spilledFiles.begin(), spilledFiles.end());
+    EXPECT_EQ(spilledFileSet.size(), spilledFiles.size());
+    EXPECT_EQ(expectedSpilledFiles, spilledFileSet.size());
+    // Verify the spilled file exist on file system.
+    fs = filesystems::getFileSystem(tempDirectory->path, nullptr);
+    for (const auto& spilledFile : spilledFileSet) {
+      auto readFile = fs->openFileForRead(spilledFile);
+      EXPECT_NE(readFile.get(), nullptr);
+    }
+
+    for (auto partition = 0; partition < state.maxPartitions(); ++partition) {
+      auto merge = state.startMerge(partition, nullptr);
+      // We expect 10 * 20000 rows in dense increasing order.
+      for (auto i = 0; i < 200'000; ++i) {
+        auto stream = merge->next();
+        ASSERT_NE(nullptr, stream);
+        EXPECT_EQ(
+            i,
+            stream->current()
+                .childAt(0)
+                ->asUnchecked<FlatVector<int64_t>>()
+                ->valueAt(stream->currentIndex()));
+        EXPECT_EQ(
+            i, stream->decoded(0).valueAt<int64_t>(stream->currentIndex()));
+
+        stream->pop();
+      }
+      ASSERT_EQ(nullptr, merge->next());
+    }
+    // Both spilled bytes and files stats are cleared after merge read.
+    EXPECT_EQ(0, state.spilledBytes());
+    EXPECT_EQ(0, state.spilledFiles());
+  }
+  // Verify the spilled file has been removed from file system after spill state
+  // destruction.
+  for (const auto& spilledFile : spilledFiles) {
+    EXPECT_ANY_THROW(fs->openFileForRead(spilledFile));
+  }
+}
+
+TEST_F(SpillTest, spillStateWithSmallTargetFileSize) {
```

Comment:
It appears that there is a good amount of code duplication between this test and spillState. Would it make sense to refactor?